djangocms_versioning/admin.py

```diff
@@ -667,6 +699,8 @@ def revert_view(self, request, object_id):
                 request, "djangocms_versioning/admin/revert_confirmation.html", context
             )
         else:
+            # TODO: What's going on here? Why are we allowing both the
+            # archiving of a draft and the deletion of it straight after?
```

Comment:
I don't see that it does that. The logic forks here depending on GET parameters. It does either an 'archive' or 'discard'.

Reply:
```
            if draft_version and request.POST.get("archive"):
                draft_version.archive(request.user)
            if draft_version and request.POST.get("discard"):
                draft_version.delete()
```
Based on this, if a url had both archive and discard in the url, both of these would happen.

Reply:
Yes that's true, although I don't imagine that the UI would create such a scenario. What do you suggest?

Reply:
Writing a test and changing the second if to an elif ;)

Reply:
https://divio-ch.atlassian.net/browse/FIL-1003